src/Client/OfficeInterop/OfficeInterop.fs

```diff
@@ -1038,6 +1038,80 @@ let addOutputColumn (excelTable:Table) (arcTable:ArcTable) (newBB:CompositeColum
 
         loggingList
 
+let addBuildingBlock (excelTable:Table) (arcTable:ArcTable) (newBB:CompositeColumn) (headerRange:Excel.Range) (selectedRange:Excel.Range) =
+
+    let rowCount = arcTable.RowCount + 1
+
+    let buildingBlockCells = Spreadsheet.CompositeColumn.toStringCellColumns newBB
+
+    let targetIndex =
+        let excelIndex = rebaseIndexToTable selectedRange headerRange
+
+        let headers = arcTable.ToExcelValues().[0] |> Array.ofSeq |> Array.map (fun header -> header.ToString())
+
+        if excelIndex < (float) (headers.Length - 1) then 
+
+            //We want to start looking after the chosen column in order to ignore the current column because it could be a main column
+            let headers = headers.[(int) excelIndex + 1..]
+
+            let mainColumNames =
+                // Get all cases of the union
+                Microsoft.FSharp.Reflection.FSharpType.GetUnionCases(typeof<CompositeHeader>)
+                |> Array.map (fun case -> case.Name)
+
+            let indices =
+                mainColumNames
+                |> Array.map (fun name ->
+                                headers
+                                |> Array.tryFindIndex (fun header -> header.StartsWith(name)))
+                                |> Array.where (fun index -> index.IsSome)
+                                |> Array.map (fun index -> index.Value)
+
+            if indices.Length > 0 then
+
+                let indices =
+                    indices
+                    |> Array.sortBy(fun index -> index)
+
+                //The +1 makes sure, we add the column after the column we have currently chosen
+                excelIndex + (float indices.[0]) + 1.
+
+            //The +1 makes sure, we add the column after the column we have currently chosen
+            else excelIndex + 1.
+        else
+            //Add the new building block to the end of the table
+            -1.
+
+    let headers =
+        headerRange.values[0]
+        |> List.ofSeq
+        |> List.map (fun header -> header.ToString())
+
+    let bb = 
```

Comment:
we do nothing with this? Use for loop and call sideffect excel functions inside. Therefore no new lists get instantiated for improved performance.

Reply:
You may also use `List.iteri`

Reply:
fixed, removed bb
Was a rest from last testing sry